src/gui/search/searchsortmodel.cpp

```diff
@@ -47,8 +119,47 @@ bool SearchSortModel::lessThan(const QModelIndex &left, const QModelIndex &right
         if (Utils::String::naturalSort(vL.toString(), vR.toString(), res))
             return res;
 
-        return QSortFilterProxyModel::lessThan(left, right);
+        return base::lessThan(left, right);
+    }
+
+    return base::lessThan(left, right);
+}
+
+bool SearchSortModel::filterAcceptsRow(int sourceRow, const QModelIndex &sourceParent) const
+{
+    if (m_isNameFilterEnabled && !m_searchTerm.isEmpty()) {
+        QString name = sourceModel()->data(sourceModel()->index(sourceRow, NAME)).toString();
+        for (const QString& word: m_searchTermWords) {
+            int i = name.indexOf(word, 0, Qt::CaseInsensitive);
+            if (i == -1) {
+                return false;
+            }
+        }
+    }
+
+    if (m_minSize > 0 || m_maxSize >= 0) {
+        qlonglong size = sourceModel()->data(sourceModel()->index(sourceRow, SIZE)).toLongLong();
+        if ((m_minSize > 0 && size < m_minSize)
+            ||  (m_maxSize > 0 && size > m_maxSize)) {
+            return false;
+        }
+    }
+
+    if (m_minSeeds > 0 || m_maxSeeds >= 0) {
+        int seeds = sourceModel()->data(sourceModel()->index(sourceRow, SEEDS)).toInt();
+        if ((m_minSeeds > 0 && seeds < m_minSeeds)
+            ||  (m_maxSeeds > 0 && seeds > m_maxSeeds)) {
+            return false;
+        }
+    }
+
+    if (m_minLeechs > 0 || m_maxLeechs >= 0) {
+        int leechs = sourceModel()->data(sourceModel()->index(sourceRow, LEECHS)).toInt();
```

Comment:
The correct plural of the word leech is "leeches". Please correct all variations of it.